File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        String arr[] ={"zero","one","two","three","four","five","six","seven","eight","nine"};
        String num[] = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9"};
        for(int i=0; i<10; i++){
            s = s.replace(arr[i] , num[i]);
        }
        int output = Integer.parseInt(s);
        return output;
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.cpp
#include <string>
#include <vector>
#include<iostream>
#include <bits/stdc++.h>
using namespace std;

int solution(string s) {
    vector<string> arr = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    int answer = 0;
    for(int i=0; i<10; i++){
        size_t pos = s.find(arr[i]);
        while(pos != string::npos){
            s.replace(pos, arr[i].length(), to_string(i));
            pos = s.find(arr[i], pos+1);
        }
    }
    answer = stoi(s);
    return answer;
}